### PR TITLE
Update network_sockets.md

### PR DESCRIPTION
**Pull request by AnotherButler**
-Fix broken link and spelling in code comments
-Change passive to active voice
-Clarify sentences

### CODE REVIEW

File: docs/APIs/communication/network_sockets.md [hunk 1]
```diff
@@ -30,11 +30,11 @@ Existing network interfaces:
...
 
 ## Network errors
 
-The convention of the network-socket API is for functions to return negative error codes to indicate failure. On success, a function may return zero or a non-negative integer to indicate the size of a transaction. On failure, a function must return a negative integer, which should be one of the error codes in the `nsapi_error_t` enum ([here](https://github.com/mbedmicro/mbed/blob/master/features/net/network-socket/nsapi_types.h#L27)):
+The convention of the network-socket API is for functions to return negative error codes to indicate failure. On success, a function may return zero or a non-negative integer to indicate the size of a transaction. On failure, a function must return a negative integer, which should be one of the error codes in the `nsapi_error_t` enum ([here](https://github.com/ARMmbed/mbed-os/blob/master/features/netsocket/nsapi_types.h)):
```
Comment (iriark01) [inline docs/APIs/communication/network_sockets.md]: Can you check with someone if https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.3/api/group__netsocket.html#gac21eb8156cf9af198349069cdc7afeba is correct?

File: docs/APIs/communication/network_sockets.md [hunk 2]
```diff
@@ -63,6 +63,6 @@ typedef enum nsapi_error {
...
 
-To allow efficient use of nonblocking operations, the socket classes provide an ``attach`` member function to register a callback on socket state changes. The callback will be called when the socket can successfully receive, send or accept, or when an error occurs. The callback may be called spuriously without reason.
+To allow efficient use of nonblocking operations, the socket classes provide an ``attach`` member function to register a callback on socket state changes. When the socket can successfully receive, send or accept, or when an error occurs, the system triggers a callback. It may call the callback spuriously without reason.
 
-The callback may be called in an interrupt context and should not perform expensive operations such as receiving and sending calls.
+The callback may be called in interrupt context and should not perform operations such as receiving and sending calls. Instead, take precautions accordingly and do not make any read or write calls until it is on a thread.
```
Comment (iriark01) [inline docs/APIs/communication/network_sockets.md]: Not wild about a generic "take precautions". Is there anything other than waiting until the callback is on a thread? If not, then just talk about waiting for the thread. If there is anything else, we should probably explain it or reference some other explanation.

File: docs/APIs/communication/network_sockets.md (hunk 1 above)
Comment (AnotherButler) [inline docs/APIs/communication/network_sockets.md]: @theotherjimmy Could you please confirm whether this link is correct?

Comment (theotherjimmy) [inline docs/APIs/communication/network_sockets.md]: https://github.com/ARMmbed/mbed-os/blob/master/features/netsocket/nsapi_types.h#L37 is exactly correct, but without the line number is good enough.

Comment (iriark01) [inline docs/APIs/communication/network_sockets.md]: We're asking about using the link to the rendered doxy on our docs site rather than sending to GitHub

Comment (theotherjimmy) [inline docs/APIs/communication/network_sockets.md]: Yeah, the link you provided looks correct.

**Comment by iriark01** [comment]
Can I move this to 5.4 and merge it?

**Comment by AnotherButler** [comment]
Yes, I think so. Shall I?

**Comment by iriark01** [comment]
Yep

**Pull request merged**